BackEnd/src/main/java/com/unosquare/admin_core/back_end/controller/BaseController.java

```diff
@@ -51,12 +58,58 @@
                 EventDTO newEvent = modelMapper.map(date, EventDTO.class);
                 modelMapper.map(createEventViewModel, newEvent);
                 newEvent.setEventTypeId(eventType.getEventTypeId());
-                eventDTOS.add(newEvent);
+                if (eventType.getEventTypeId() == EventTypes.ANNUAL_LEAVE.getEventTypeId()) {
+                    String groupId = UUID.randomUUID().toString();
+                    newEvent.setGroupId(groupId);
+                    splitEventIfFallsOnAWeekend(newEvent, newEvent.getEndDate(), eventDTOS);
+                } else {
+                    eventDTOS.add(newEvent);
+                }
             }
-
             eventService.saveEvents(eventDTOS.toArray(new EventDTO[0]));
         }
 
         return responses;
     }
+
+    private void splitEventIfFallsOnAWeekend(EventDTO event, LocalDate originalEndDate, List<EventDTO> eventDTOS) {
+        List<LocalDate> dateRange = getDatesRange(event.getStartDate(), event.getEndDate());
+        for (LocalDate eventDate : dateRange) {
+            if (eventDate.getDayOfWeek() == DayOfWeek.SATURDAY) {
+                // Set end date
+                event.setEndDate(eventDate.minusDays(1));
+                eventDTOS.add(event);
+                // Create new event
+                EventDTO nextEvent = mapEventDto(event, originalEndDate, eventDate, 2);
+                // Check again
+                splitEventIfFallsOnAWeekend(nextEvent, nextEvent.getEndDate(), eventDTOS);
+                break;
+            }
+        }
+
+        if (event.getEndDate() == originalEndDate){
+            eventDTOS.add(event);
+        }
+    }
+
+    private EventDTO mapEventDto(EventDTO priorEvent, LocalDate originalEndDate, LocalDate eventDate, int plusDays) {
+        EventDTO nextEvent = new EventDTO();
```

Comment:
Would be nice if modelmapper tool care of a lot of this. EventDto to EventDto mapping.